### PR TITLE
[WASM] Add a JsValue class

### LINKED ISSUE #808: How should JS values bind into Python?
This does not require knowledge of the WASM crate's internals, only a high level knowledge of Python and JS objects.

A couple questions about adding a `JsValue` class to WASM:

* When passing in a JS value into python, where should the line be between `JsValue` and builtin Python type? e.g. JS strings and numbers should be converted to their corresponding Python primitives, but what about arrays/lists? `TypedArrays`/`bytes`/`bytearray`s?
* Should functions be `JsValue`s too? Should JsValue have a `__call__` method?
* Should we have a `__getattr__` method on `JsValue`?
    * Should we throw an error when a `JsValue` doesn't have an attribute, or just return `None`?
* How should we transfer `kwargs` to JS functions? Right now it provides them as the `this` variable, but should we change that?
* How should errors transfer/convert between the two languages?

Not all of these questions are necessarily hard to answer, but it's good to have them on the table.

### PR DESCRIPTION
**Pull request by coolreader18**
Attempt number 2, after #848. Here's the design for this implementation:

- `_js` module - contains the `JsValue` class which wraps a `wasm_bindgen::JsValue` and provides methods like `get_prop()`, `set_prop()`, and `typeof()`.
- `_window` - provides a single export, a `JsValue` wrapping `window`. Can be disabled by the JS code using `rustpython_wasm`.
- `browser` - a (eventually) Python frozen module that imports `window` from `_window` and uses it to provide all the functionality that the `browser` native module does now.
- (eventually) `jslib` - a frozen module that provides a JsProxy similar in functionality to the `JsValue` discussed in #808 among some other things (like a Promise wrapper).

Closes #808.

### CODE REVIEW

**Comment by coolreader18** [comment]
Ok, I think this is good for review as an initial implementation; everything works that did before, and although it's not all converted to the new system, I think that's best done in follow-up PRs.

File: vm/src/lib.rs [hunk 1]
```diff
@@ -66,3 +66,8 @@ mod vm;
 pub use self::exceptions::print_exception;
 pub use self::vm::VirtualMachine;
 pub use rustpython_compiler::*;
+
+#[doc(hidden)]
+pub mod __exports {
```
Comment (windelbouwman) [inline vm/src/lib.rs]: I think the same can be achieved by using `extern crate bincode`. Cargo will make sure that both the vm and the wasm crate use the same version of the bincode crate, so no need to import the bincode crate via the `rustpython_vm`.

File: vm/src/stdlib/mod.rs [hunk 2]
```diff
@@ -34,6 +34,7 @@ use crate::pyobject::PyObjectRef;
 pub type StdlibInitFunc = Box<dyn Fn(&VirtualMachine) -> PyObjectRef>;
 
 pub fn get_module_inits() -> HashMap<String, StdlibInitFunc> {
+    #[allow(unused_mut)]
```
Comment (windelbouwman) [inline vm/src/stdlib/mod.rs]: Why was this added? We're returning an immutable HashMap here right?

**Comment by windelbouwman** [comment]
@rmliddle could you take a view on this change?

File: vm/src/stdlib/mod.rs (hunk 2 above)
Comment (coolreader18) [reply vm/src/stdlib/mod.rs]: A little bit downwards there are some `#[cfg]`s that conditionally add modules to the map, and if your target matches neither of those `cfg`s, you get an unused mut warning.

Comment (windelbouwman) [reply vm/src/stdlib/mod.rs]: Ah, I understand now. Thanks!

File: vm/src/lib.rs (hunk 1 above)
Comment (coolreader18) [reply vm/src/lib.rs]: I did this here because this pattern is used by other macros in the ecosystem, here are a few examples:
- https://docs.rs/serde/1.0.92/src/serde/lib.rs.html#236-242
- https://docs.rs/wasm-bindgen/0.2.47/src/wasm_bindgen/lib.rs.html#801
- https://docs.rs/darling/0.9.0/src/darling/lib.rs.html#84-95
